zcash_client_memory/src/types/block.rs

```diff
@@ -0,0 +1,154 @@
+use std::{
+    cmp::Ordering,
+    collections::{HashMap, HashSet},
+};
+
+use zcash_client_backend::wallet::NoteId;
+use zcash_primitives::{block::BlockHash, consensus::BlockHeight, transaction::TxId};
+use zcash_protocol::memo::MemoBytes;
+
+/// Internal wallet representation of a Block.
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) struct MemoryWalletBlock {
+    pub(crate) height: BlockHeight,
+    pub(crate) hash: BlockHash,
+    pub(crate) block_time: u32,
+    // Just the transactions that involve an account in this wallet
+    pub(crate) _transactions: HashSet<TxId>,
+    pub(crate) _memos: HashMap<NoteId, MemoBytes>,
```

Comment:
This is an odd representation given that `_transactions` doesn't keep the transaction index information, but I guess it's fine.